src/v2/guide/migration-vue-router.md

```diff
@@ -236,6 +236,23 @@ if (route.meta.requiresAuth) {
 </div>
 {% endraw %}
 
+### [] Syntax for Array of Route Query <sup>removed</sup>
+
+In URL, [] syntax for array of query is removed. For example, `/foo?users[]=Tom&users[]=Jerry` will be `/foo?users=Tom&users=Jerry` now.
+
+Since vue-router will treat `?users=Tom` as `query.users == 'Tom'`, unless you are for sure that the array will have more than one element, you should check and convert the query to array if necessary.
+
+One practical way is to add the following kind of watcher:
+
+```javascript
+'$route.query.users': {
+  handler(val) {
+    this.$route.query.users = Array.isArray(val) ? val : [val]
```

Comment:
To my opinion, when migrated from vue-router 0.7 and when the user is very sure that the query is an array, the watcher will not break the consistency of $route object.

This kind of watcher will minimize the migration cost especially when the query array is used in multiple places.